### ISSUE TITLE
IsEmpty() not working on index match

### ISSUE THREAD

**Issue by aorsten** [fauna/faunadb-js#303]
The following code:
```js
q.Let(
  {
    hashMatch: q.Match(
      q.Index("MyIndex"),
      "someHash"
    )
  },
  {
    empty: q.IsEmpty(q.Var('hashMatch')),
    values: q.Paginate(q.Var('hashMatch'))
  }
)
```
Returns: 
```
{
  empty: true,
  values: { data: [ 'found-a-value' ] } 
}
```
The code runs as expected in the Fauna Shell, but IsEmpty fails in the JS driver.
So basically IsEmpty on a SetRef does not work properly.

**Comment by marrony** [comment]
> The code runs as expected in the Fauna Shell, but IsEmpty fails in the JS driver.
does that means `IsEmpty` worked on Fauna Shell, but not on JS driver?

**Comment by aorsten** [comment]
Yes, I believe so. I copied my exact code from js to the fauna shell, and there it worked just fine.

**Comment by aorsten** [comment]
I solved it for myself, since I basically just needed to wrap my paginate in a select and yield a default value when item 0 was missing. (sent on mobile)

**Comment by parkhomenko** [comment]
hi @aorsten 
sorry for the late reply

I've run your query, first with a simple admin secret and then using ABAC, all the options worked fine for me.
Here is how I tested with ABAC:
1. I have created a collection "Users" with "email" fiels and a "password" as credentials:
`Create(
  Collection("Users"),
  {
    data: {
      email: "john@empire.com"
    },
    credentials: {
      password: "donttellanyone"
    }
  }
)`
2. Then a new role with permissions to read the index (it just tells that anyone from "Users" collection can read the "MyIndex"):
`CreateRole({
  name: "User",
  membership: {
    resource: Collection("Users")
  },
  privileges: [
    {resource: Index("MyIndex"), actions: {read: true}}
  ]
})`
3. Created an in index for a "Login" command:
`CreateIndex({
  name: "Users_by_email",
  source: Collection("Users"),
  terms: [{field: ["data", "email"]}],
  unique: true
})`
4. Invoked the "Login" command to receive a temporary token:
`Login(
  Match(Index("Users_by_email"), "john@empire.com"),
  {
    password: "donttellanyone",
    ttl: TimeAdd(Now(), 3, 'hour')
  }
)`
5. After those steps I was able to run your query in js client:
`const client = new faunadb.Client({
    secret: '<secret_received_from_login_command>'
});
const query = q.Let(
        {
            hashMatch: q.Match(
                q.Index("MyIndex"),
                "Laptops"
            )
        },
        {
            empty: q.IsEmpty(q.Var('hashMatch')),
            values: q.Paginate(q.Var('hashMatch'))
        }
    )
client.query(query).then(res => console.log(res));
`
